File: scratch/search_patterns.php
<?php
// scratch/search_patterns.php

$rootDir = dirname(__DIR__);

$patterns = [
    'users_queries' => '/FROM\s+["\']?\s*\.\s*TBL_USERS/i',
];

$results = [];
foreach ($patterns as $name => $regex) {
    $results[$name] = [];
}

$excludeDirs = ['.git', 'vendor', 'laravel', 'node_modules', 'scratch'];

$scan = null;
$scan = function($dir) use (&$results, $patterns, $excludeDirs, $rootDir, &$scan) {
    $files = scandir($dir);
    foreach ($files as $file) {
        if ($file === '.' || $file === '..') continue;
        $path = $dir . '/' . $file;
        if (is_dir($path)) {
            if (in_array($file, $excludeDirs)) continue;
            $scan($path);
        } else {
            if (pathinfo($path, PATHINFO_EXTENSION) !== 'php') continue;
            $content = file_get_contents($path);
            $lines = explode("\n", $content);
            foreach ($lines as $idx => $line) {
                foreach ($patterns as $name => $regex) {
                    if (preg_match($regex, $line)) {
                        $results[$name][] = [
                            'file' => str_replace($rootDir . '/', '', $path),
                            'line' => $idx + 1,
                            'content' => trim($line),
                        ];
                    }
                }
            }
        }
    }
};

$scan($rootDir);

echo "--- RESULTS ---\n";
foreach ($results as $name => $matches) {
    if (!in_array($name, ['users_queries'])) continue;
    echo "\n[" . strtoupper($name) . "] Matches count: " . count($matches) . "\n";
    foreach ($matches as $m) {
        if ($m['file'] !== 'controllers/logic.php') continue;
        echo "  " . $m['file'] . ":" . $m['line'] . " -> " . $m['content'] . "\n";
    }
}
?>

File: scratch/search_patterns.py
import os
import re

root_dir = r"g:\22 june sheba"

patterns = {
    'display_errors': re.compile(r"display_errors"),
    'error_reporting': re.compile(r"error_reporting"),
    'md5': re.compile(r"\bmd5\("),
    'debug_logging': re.compile(r"file_put_contents\([^,]+debug"),
    'var_dump': re.compile(r"\bvar_dump\b|\bprint_r\b"),
}

results = {k: [] for k in patterns}

exclude_dirs = {'.git', 'vendor', 'laravel', 'node_modules', 'scratch'}

for root, dirs, files in os.walk(root_dir):
    # filter out excluded directories
    dirs[:] = [d for d in dirs if d not in exclude_dirs]
    for file in files:
        if not file.endswith('.php'):
            continue
        filepath = os.path.join(root, file)
        try:
            with open(filepath, 'r', encoding='utf-8', errors='ignore') as f:
                content = f.read()
                lines = content.split('\n')
                for idx, line in enumerate(lines):
                    for name, pat in patterns.items():
                        if pat.search(line):
                            results[name].append((filepath, idx + 1, line.strip()))
        except Exception as e:
            print(f"Error reading {filepath}: {e}")

print("--- RESULTS ---")
for name, matches in results.items():
    print(f"\n[{name.upper()}] Matches count: {len(matches)}")
    for filepath, line_num, line_content in matches[:100]: # limit to 100 for readability
        rel_path = os.path.relpath(filepath, root_dir)
        print(f"  {rel_path}:{line_num} -> {line_content}")
